src/app/projects/project/SlotRequestModal.js

```diff
@@ -184,6 +184,34 @@ export default function SlotRequestModal({
     setProcessing(false);
   };
 
+  const handleJoinSlot = async () => {
+    setProcessing(true);
+    setError();
+
+    try {
+      // check if the slot isn't already full
+      if (Object.keys(slot?.persons || {}).length < MAX_PARTNERS) {
```

Comment:
Instead of checking if the slot is full after the user has clicked a button, can we hide the button if the slot is full?